src/core.jl

```diff
@@ -42,36 +43,57 @@ struct Shortcut{T}
     shortcut_res::Matrix{T}
 end
 
+abstract type Solver end
+
+struct CholmodSolver <: Solver
+    bs::Int
+end
+
+struct AMGSolver <: Solver
+end
+
+struct MKLPardisoSolver <: Solver
+    bs::Int
+end
 """
 Core kernel of Circuitscape - used to solve several pairs
 
 Input:
-* data::GraphData
+* data::GraphProblem
 """
-function single_ground_all_pairs(data::GraphData{T,V}, flags, cfg, log = true) where {T,V}
+function single_ground_all_pairs(prob::GraphProblem{T,V,W}, flags, cfg, log = true) where {T,V,W}
+    solve(prob, prob.solver, flags, cfg, log)
+end
 
-    if flags.solver in AMG
+function get_solver(cfg)
+    s = cfg["solver"]
+    if s in AMG
         csinfo("Solver used: AMG accelerated by CG")
-        amg_solver_path(data, flags, cfg, log)
-    else
+        return AMGSolver()
+    elseif s in CHOLMOD
+        csinfo("Solver used: CHOLMOD")
+        bs = parse(Int, cfg["cholmod_batch_size"])
+        return CholmodSolver(bs)
+    elseif s in MKLPARDISO
         csinfo("Solver used: CHOLMOD")
```

Comment:
Shouldn't this be `csinfo("Solver used: MKLPardiso")`?

Reply:
Yes, correct.